activerecord/lib/active_record/internal_metadata.rb

```diff
@@ -0,0 +1,56 @@
+require 'active_record/scoping/default'
+require 'active_record/scoping/named'
+
+module ActiveRecord
+  # This class is used to create a table that keeps track of values and keys such
+  # as which environment migrations were run in.
+  class InternalMetadata < ActiveRecord::Base
+    class << self
+      def primary_key
+        "key"
+      end
+
+      def table_name
+        "#{table_name_prefix}#{ActiveRecord::Base.internal_metadata_table_name}#{table_name_suffix}"
+      end
+
+      def index_name
+        "#{table_name_prefix}unique_#{ActiveRecord::Base.internal_metadata_table_name}#{table_name_suffix}"
+      end
+
+      def store(hash)
+        hash.each do |key, value|
+          first_or_initialize(key: key.to_s).update_attributes!(value: value)
+        end
+      end
```

Comment:
That's fair. To me, absent a more comprehensive suite of hash-like methods, `[]`/`[]=` are just the "right" way of spelling "get/set value using a key", though.

And this is so far away from anything a user would ever touch, it seems fine to go with whatever makes our code look neatest.